library/src/main/java/com/google/maps/android/data/Renderer.java

```diff
@@ -94,9 +96,30 @@
 
     private HashMap<KmlGroundOverlay, GroundOverlay> mGroundOverlayMap;
 
-    private final ArrayList<String> mMarkerIconUrls;
+    private final Set<String> mMarkerIconUrls;
+
+    /**
+     * Map of image URL to map of scale factor to BitmapDescriptors for point marker icons
+     *
+     * BitmapDescriptors are cached to avoid creating new BitmapDescriptors for each individual
+     * usage of a Bitmap. Each BitmapDescriptor copies the Bitmap it's created from.
+     */
+    private Map<String, Map<Double, BitmapDescriptor>> mMarkerImagesCache;
```

Comment:
The probability of missing the cache can be pretty high having `Double` as a key for `Map`. I suggest storing this as `String` instead and have up to a certain precision level (e.g. 4 decimal places?)

Reply:
I had thought about this briefly and concluded that while a `Double` may not be able to perfectly represent all parsed decimal strings, 0.1 for example, that at least the parsed string double value will be equal for the scale value used to cache the image and scale value used to access the cached scaled image. In other words, a double representation of 0.1 will equal another double representation of 0.1, both obtained from the same parse routine. In fact, the same double `mScale` variable from a `KmlStyle` is often being used to both cache and retrieve the cached images.

However, while I made this conclusion based on the current implementation, I do see how a bug could possibly be introduced in the future, if style scales were introduced from another source somehow, possibly with a lower float precision.

I made this change to use the string formatted scale value: https://github.com/googlemaps/android-maps-utils/pull/609/commits/f9bd8baac9ce104b605e1221edd99d79ba0d7df2